code/modules/mob/observer/eye/landing_eye/landing_eye.dm

```diff
@@ -0,0 +1,93 @@
+#define LANDING_VIEW 12
+
+/mob/observer/eye/landing
+	name = "Landing Eye"
+	desc = "A visual projection used to assist in the landing of a shuttle."
+	name_sufix = "Landing Eye"
+	var/datum/shuttle/autodock/shuttle
+	var/list/landing_images = list()
+
+/mob/observer/eye/landing/Initialize(var/mapload, var/shuttle_tag)
+	shuttle = SSshuttle.shuttles[shuttle_tag]
+	// Generates the overlay of the shuttle on turfs.
+	var/turf/origin = get_turf(src)
+	for(var/area/A in shuttle.shuttle_area)
+		for(var/turf/T in A)
+			var/image/I = image('icons/effects/alphacolors.dmi', origin, "red")
+			// Record the offset of the turfs from the eye. The eye is where the shuttle landmark will be placed, so the resultant images will reflect actual landing.
+			var/x_off = T.x - origin.x
+			var/y_off = T.y - origin.y
+
+			I.loc = locate(origin.x + x_off, origin.y + y_off, origin.z)
```

Comment:
Should check world bounds when doing this, otherwise you're generating images nobody will ever see.